pkg/oc/originpolymorphichelpers/logsforobject.go

```diff
@@ -39,6 +42,30 @@ func NewLogsForObjectFn(delegate polymorphichelpers.LogsForObjectFunc) polymorph
 				return nil, err
 			}
 			return appsmanualclient.NewRolloutLogClient(appsClient.Apps().RESTClient(), t.Namespace).Logs(t.Name, *dopts), nil
+		case *appsapiv1.DeploymentConfig:
+			dopts, ok := options.(*appsapiv1.DeploymentLogOptions)
+			if !ok {
+				return nil, errors.New("provided options object is not a DeploymentLogOptions")
+			}
+			appsClient, err := appsclientinternal.NewForConfig(clientConfig)
+			if err != nil {
+				return nil, err
+			}
+			return appsmanualclient.NewRolloutLogClient(appsClient.Apps().RESTClient(), t.Namespace).Logs(t.Name, *dopts), nil
```

Comment:
@deads2k not sure how to handle this - do we have a rollout client (or buildlogClient for the case below) whose Logs method receives versioned options?

Reply:
i made an external version for this here https://github.com/openshift/origin/pull/20362